lib/ProjectSystemEpic/PackageFeature/Epics/Packages.js

```diff
@@ -3,14 +3,21 @@
 
 import Rx from "rxjs";
 import { packagesRefreshed } from "../Actions/PackagesRefreshed";
+import { packageRefreshCompleted } from "../Actions/PackageRefreshCompleted";
+import type { Package } from "../Types/types";
 
-const packagesEpic = findPackages => action$ => {
+type packageFinder = (
+  rootPath: string,
+  plugins: Array<Plugin>,
+) => Rx.Observable<Array<Package>>;
+
+const packagesEpic = (findPackages: packageFinder) => (action$: Observable) => {
   return action$
     .ofType("REFRESH_PACKAGES")
-    .groupBy(a => a.payload.rootPath, null, () => Rx.Observable.timer(3000))
-    .map(g => g.buffer(Rx.Observable.timer(3000)))
+    .groupBy(a => a.payload.rootPath, null, () => Rx.Observable.timer(300))
+    .map(g => g.buffer(Rx.Observable.timer(300)))
```

Comment:
why going from 3000 to 300 ?

Reply:
Two reasons:

- Most of these events happen very close to each other, so having 3s of buffering is overkill.

- While the 3s delay isn't too visible when starting Atom, it's very visible when reloading a package (eg saving changes in package.json).

Reply:
hmmm ok